Sources/KSCrashRecording/KSCrash.m

```diff
@@ -221,65 +221,26 @@ - (NSDictionary *)systemInfo
 - (BOOL)installWithConfiguration:(KSCrashConfiguration *)configuration error:(NSError **)error
 {
     self.configuration = [configuration copy] ?: [KSCrashConfiguration new];
-    NSString *installPath = configuration.installPath ?: getDefaultInstallPath();
-    KSCrashInstallErrorCode result =
-        kscrash_install(self.bundleName.UTF8String, installPath.UTF8String, [self.configuration toCConfiguration]);
+    NSString *installPath = configuration.installPath ?: kscrash_getDefaultInstallPath();
 
-    if (result != KSCrashInstallErrorNone) {
-        if (error != NULL) {
-            *error = [self errorForInstallErrorCode:result];
-        }
+    KSCrashReportStore *reportStore = [KSCrashReportStore storeWithPath:installPath error:error];
+    if (reportStore == nil) {
         return NO;
     }
 
-    return YES;
-}
-
-- (BOOL)setupReportStoreWithPath:(NSString *)installPath error:(NSError **)error
-{
     KSCrashInstallErrorCode result =
-        kscrash_setupReportsStore(self.bundleName.UTF8String, (installPath ?: getDefaultInstallPath()).UTF8String);
-
+        kscrash_install(self.bundleName.UTF8String, installPath.UTF8String, [self.configuration toCConfiguration]);
     if (result != KSCrashInstallErrorNone) {
         if (error != NULL) {
-            *error = [self errorForInstallErrorCode:result];
+            *error = [KSCrash errorForInstallErrorCode:result];
         }
         return NO;
     }
 
+    _reportStore = reportStore;
```

Comment:
I just wanted to check whether we need to synchronise or if we can assume that all the setup logic should be performed on a single thread.

Reply:
Yeah, I don't think we have synchronization for installation logic, don't think we should worry a lot about that